common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -655,18 +635,19 @@ public static boolean isValidIpV6Address(String ipAddress) {
         return true;
     }
 
-    private static boolean isValidIp4Word(String word) {
-        char c;
-        if (word.length() < 1 || word.length() > 3) {
+    private static boolean isValidIp4Word(CharSequence word, int start, int end) {
+        int len = end - start;
+        char c0, c1, c2;
+        if (len < 1 || len > IPV4_MAX_CHAR_BETWEEN_SEPARATOR || (c0 = word.charAt(start)) < '0') {
             return false;
         }
-        for (int i = 0; i < word.length(); i ++) {
-            c = word.charAt(i);
-            if (!(c >= '0' && c <= '9')) {
-                return false;
-            }
+        if (len == IPV4_MAX_CHAR_BETWEEN_SEPARATOR) {
+            return (c1 = word.charAt(start + 1)) >= '0' &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (9) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1067)